DEVELOPMENT.md

```diff
@@ -1,179 +1,198 @@
-   - OPENAI_API_KEY - an [OpenAI API Key](https://platform.openai.com/api-keys) to use for OpenAI unit testing
-   - OLLAMA_HOST - the URL to the Ollama instance to use for testing (e.g. `http://host.docker.internal:11434`)
-   - ANTHROPIC_API_KEY - an [Anthropic API Key](https://docs.anthropic.com/en/docs/quickstart#set-your-api-key) to use for Anthropic unit testing
-   - COHERE_API_KEY - a [Cohere API Key](https://docs.cohere.com/docs/rate-limits) to use for Cohere unit testing
 
-2. If you have made changes to the source, from a docker shell, (re)install the extension
+## Test pgai
 
+The [tests](./tests) directory contains the psql scripts that run unit tests. The tests are driven by
+[test.sql](./tests/test.sql).
+
+To setup the pgai tests:
+
+1. In `<pgai_install_dir>/.env`, add the variables for the framework you want to test:
+   - **OpenAI**:
+      - ENABLE_OPENAI_TESTS - set to `true` to enable OpenAI unit tests.
+      - OPENAI_API_KEY - an [OpenAI API Key](https://platform.openai.com/api-keys) for OpenAI unit testing.
+   - **Ollama**:
+      - ENABLE_OLLAMA_TESTS - set to `true` to enable Ollama unit tests.
+      - OLLAMA_HOST - the URL to the Ollama instance to use for testing. For example, `http://host.docker.internal:11434`.
+   - **Anthropic**:
+      - ENABLE_ANTHROPIC_TESTS - set to `true` to enable Anthropic unit tests.
+      - ANTHROPIC_API_KEY - an [Anthropic API Key](https://docs.anthropic.com/en/docs/quickstart#set-your-api-key) for Anthropic unit testing.
+   - **Cohere**:
+      - ENABLE_COHERE_TESTS - set to `true` to enable Cohere unit tests.
+      - COHERE_API_KEY - a [Cohere API Key](https://docs.cohere.com/docs/rate-limits) for Cohere unit testing.
+
+2. If you have made changes to the source, from a Docker shell, install the extension:
    ```bash
+   cd <pgai_install_dir>
+   make docker-shell
    make install
    ```
+   You are in a Docker shell.
 
 3. Run the tests
 
    ```bash
    make test
    ```
 
-   This will:
-   1. drop the "test" database if it exists
-   2. create a "test" database
-   3. create a database user named "tester" if it doesn't exist
-   4. run [test.sql](./tests/test.sql) to execute the unit tests against the "test" database
+   This:
+   1. Drops the `test` database.
+   2. Creates the `test` database.
+   3. Creates the `tester` database user.
+   4. Runs [test.sql](./tests/test.sql) which execute the unit tests against the `test` database.
 
 Best practice is to add new tests when you commit new functionality.
 
-## pgai Architecture
+## The pgai build architecture
 
-The pgai extension consists of SQL files as well as Python packages. SQL files 
-are maintained in the [./sql](./sql) directory. The Python packages are 
-maintained in the [./src](./src) directory.
+pgai consists of [SQL](#sql) scripts and [Python](#python) packages.
```

Comment:
Question: The "normal" workflow is to just run `make install` from inside the docker container. That "just works". The other make targets do more specific things for flexibility/speed, but `make install` does it all. Should we note that here?

Reply:
Added a section.